src/index.js

```diff
@@ -49,73 +50,117 @@ function createCommand(queryName, implementationName) {
-            // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
-            result.selector = `${queryName}(${queryArgument(args)})`
-
-            if (result.length > 0) {
-              consoleProps.yielded = result.toArray()
-            }
-
-            return result
-          }
-
-          const resolveValue = () => {
-            // retry calling "getValue" until following assertions pass or this command times out
-            return Cypress.Promise.try(getValue).then(value => {
-              return cy.verifyUpcomingAssertions(value, waitOptions, {
-                onRetry: resolveValue,
-              })
-            })
-          }
+      const getValue = () => {
+        const value = commandImpl(win.document)
+
+        const result = Cypress.$(value)
+        if (value && options._log) {
+          options._log.set('$el', result)
+        }
 
-          if (queryRegex.test(queryName)) {
-            // For get* queries, do not retry
-            return getValue()
-          }
+        // Overriding the selector of the jquery object because it's displayed in the long message of .should('exist') failure message
+        // Hopefully it makes it clearer, because I find the normal response of "Expected to find element '', but never found it" confusing
+        result.selector = getSelector()
 
-          return resolveValue().then(subject => {
-            // Remove the error that occurred because it is irrelevant now
-            if (consoleProps.error) {
-              delete consoleProps.error
-            }
+        consoleProps.elements = result.length
+        if (result.length === 1) {
+          consoleProps.yielded = result.toArray()[0];
+        } else if (result.length > 0) {
+          consoleProps.yielded = result.toArray();
+        }
 
-            return subject
-          })
+        if (result.length > 1 && !/all/i.test(queryName)) {
+          // Is this useful?
+          throw Error(`Found multiple elements with the text: ${queryArgument(args)}`)
+        }
+
+        return result
+      }
+
+      if (queryRegex.test(queryName)) {
+        // make the timeout extremely short to ensure `query*` commands pass or fail instantly
+        options.timeout = 0
+      }
+
+      let error
+      // Errors will be thrown by @testing-library/dom, but a query might be followed by `.should('not.exist')`
+      // We just need to capture the error thrown by @testing-library/dom and return an empty jQuery NodeList
+      // to allow Cypress assertions errors to happen naturally. If an assertion fails, we'll have a helpful
+      // error message handy to pass on to the user
+      const catchQueryError = err => {
+        error = err
+        const result = Cypress.$()
+        result.selector = getSelector()
+        return result
+      }
```

Comment:
This is a clever hack for both useful errors and tie into the Cypress assertion system. Cypress DOM assertions expect a jQuery NodeList and have special handling for `.should('exist')` and others. Even if `@testing-library/dom` throws an error, we should still return an empty jQuery NodeList so that failures in this library are consistent with other Cypress DOM-based commands.